\documentclass[12pt]{article}
\usepackage[osf]{mathpazo}
\usepackage{ms}
\usepackage{natbib}
\usepackage{lineno}
\usepackage{graphicx}
\usepackage{caption}
\modulolinenumbers[5]
\linenumbers

\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\title{The adequacy of trait models and the rise of angiosperm functional diversity}
\author{
Matthew W. Pennell$^{1}$, William K. Cornwell$^{2}$,\\
Richard G. FitzJohn$^{3}$ \& Luke J. Harmon$^{1}$
}

\date{}
\affiliation{
\begin{center}
\textit{
$^{1}$ Department of Biological Sciences \& Institute for Bioinformatics and Evolutionary Studies, University of Idaho, Moscow, ID 83844, U.S.A.\\[0.5cm]
$^{2}$ School of BEES, University of New South Wales, Sydney, NSW 2052, Australia\\[0.5cm]
$^{3}$ Department of Biological Sciences, Macquarie University, Sydney, NSW 2109, Australia\\[0.5cm]
Email for correspondence: mwpennell@gmail.com}
\end{center}
}
\runninghead{are macroevolutionary models adequate?}


\begin{document}
\mstitlepage
\parindent=1.5em
\addtolength{\parskip}{.3em}

\begin{abstract}
Investigating evolutionary hypotheses using interspecific data necessitates the use of a statistical model of trait evolution along the phylogeny. A wide variety of such models have been proposed and fitting and comparing these is now a routine component of comparative studies. However, an often over--looked statistical question is whether the models we are using are actually capturing the relevant variation in traits. A model may be the best fit (using some model selection criterion) relative to a set of candidate models but it may still be a poor fit in an absolute sense. As the scale of comparative datasets and questions continues to increase, the question of model fit becomes even more important to consider. Here we develop a general framework for assessing the fit (or, adequacy) of quantitative trait models, based on Felsenstein's Independent Contrasts method. Our framework can be applied to arbitrarily complex trait models for which the tip values are assumed to multivariate normally distributed. We then use this approach to assess whether simple models are reasonabe descriptors of the macroevolutionary dynamics of three important angiosperm functional traits: seed size, specific leaf area, and nitrogen content. We analyzed 1xx comparative datasets across the angiosperm phylogeny and demonstrate that there is tremendous variability in the adequacy of simple models across groups, traits and time--scales. We argue that the assessment of model adequacy should be included in all comparative studies and that the framework we present can be generally applied to this end.  
\end{abstract}

\vfill

\newpage


\begin{quotation}
\noindent There are known knowns; there are things we know we don't know. There are known unknowns; that is to say, there are things that we now know we don't know. But there are also unknown unknowns --- there are things we do not know we don't know. 

---Fmr. U.S. Secretary of Defense Donald Rumsfeld
\end{quotation}

\section{Introduction}
Angiosperms are one of the most spectacular radiations in the history of the earth. The $>$300,000 species which make up the clade have diversified into a huge array of functional forms, from the grasses of the Serengeti to the trees of the Amazonian rainforest, and have come to dominate most terrestrial ecosystems. As researchers interested in macroevolution, we would like to understand how such trait diversity came to be --- the who, what, when, where and whys of deep time. These questions need to be placed in a phylogenetic context if we are to make substantiave progress towards addressing them. For example, seed size, which is a proxy for a plant's life--history strategy, varies $\sim$ 11 orders of magnitude across angiosperms \citep{Westoby1992TREE, Lord1995AmNat, Westoby2002, Moles2005, Cornwell2013} [COMPLETE EXAMPLE HERE]

The last few decades have seen a tremendous growth in statistical methods that make use of phylogenetic trees and comparative data to address macroevolutionary questions \citep[for a recent review see][]{PennellHarmon}. We can ask whether traits are evolutionary correlated with one another \citep[e.g.][]{Felsenstein1985, Grafen1989}, whether different lineages have evolved at different rates \citep[e.g.][]{Omeara2006, Eastman2011}, what the predominant mode of evolution has been \citep[e.g.][]{HansenMartins1996, Mooers1999, Harmon2010}, how the mode has varied across clades \citep[e.g.][]{ButlerKing2004, Beaulieu2012}, and how trait evolution has influenced the diversification of lineages \citep[e.g.][]{Maddison2007, FitzJohn2010}. However, all of these necessarily rely on our ability to model the evolution of traits along a phylogeny; and as eloquently articulated by \citet{Hunt2012}, any statements regarding the rate (tempo) of evolution are contigent about the particular model of change (mode).

A number of models of trait evolution have been proposed \citep[see][]{Omeara2012, PennellHarmon}. Brownian motion (BM), in which evolution proceeds via an undirected random walk such that the variance in the trait value accumulates proportional to time, is the oldest, and most commonly applied model of continuous character evolution. Originally developed as a model for estimating phylogenetic trees from allelic frequencies \citep{EC1964}, BM has been used as a more general model of phylogenetic change in phylogenetics \citep{Felsenstein1973, Thompson1975, Felsenstein1988} and is the underlying model in Felsenstein's phylogenetically independent contrasts \citep[PICs;][]{Felsenstein1985} for investigating correlated evolution. Another model is the Ornstein--Uhlenbeck process \citep[OU;][]{Hansen1997} which can be loosely described as ``evolution on a spring'' --- variance accumulates at a rate $\sigma^2$, but is ``pulled'' towards a mean value $\theta$ with some strength $\alpha$. Alternate models that have been considered include: the ``Early Burst'' \citep[EB;][]{Blomberg2003, Harmon2010, SlaterPennell} model, in which most of the evolution occurs early in the clades history; a BM model with a trend in the mean trait value \citep{Hunt2006}; models depicting jumps in phenotypic space \citep{Landis2012, Eastmanlevy}; and other variations on the models described above \citep[e.g.][]{Pagel1997, Pagel1999, ButlerKing2004, Omeara2006, Eastman2011, Beaulieu2012, SlaterMEE}.

The general procedure for using a macroevolutionary models is to first compare amongst a set of candidate models, using likelihood ratio tests or information criteria. The preferred model is then used in one of two ways: 1) make inferences directly from the model fit by interpreting the observed pattern \citep[e.g.][]{Harmon2010}; or 2) use the model to test other evolutionary hypotheses \citep[e.g.][]{Grafen1989, Freckleton2009}. In either case, there are a number of important questions that need to be considered. One is that of interpretation --- what can our models tell us about the process of evolution \citep{HansenMartins1996, Hansen2012, PennellHarmon}? Another is statistical --- is the model we are using capturing sufficient amount of the variation to address the question we are interested in? The latter question, known in statistical terminology as model adequacy, is the subject of our investigation here.

Consider again the example of seed size evolution in angiosperms. We may want to ask, for instance whether the variation in seed size we observe today accumulated early in the history of some clade of interest, which may suggest an adaptive radiation \citep{Simpson1944, Schulter2000, Yoder2010, SlaterPennell}. To do so, we could fit mulitple models of trait evolution --- an Early Burst model \citep{Blomberg2003, Harmon2010} and a BM model. We could then compare the model fits using some model selection criterion \citep[e.g. AIC;][]{Akaike1974}. Once we have the best--fitting model in hand, we would like to be able to draw inferences from it as to the general tempo of seed size in our group. But before we can do so, we would want to ensure that our model is actually capturing the relevant variation --- that is, is our model adequate?

In many statistical applications, assessing the absolute fit of the model \textit{a posteriori} is a routine procedure \citep{Gelmanbook}. Before drawing inferences about the parameters of the model, we want to know whether the model we used is adequately capturing the relevant patterns in the data. Model adequacy has been investigated for models of sequence evolution for the purposes of inferring phylogenetic trees \citep[e.g.][]{GautLewis1995, SullivanSwofford, Goldman, HuelsenbeckBull1996, SandersonKim, Bollback2002, Ripplinger2010, Lewis2013, Brown2013} and forms the basis for the Decision--Theoretic approach to model selection in phylogenetics \citep{Minin2003, Abdo2005, SullivanJoyce2005}. While there has been a great deal of discussion of the appropriateness of various models in the context of phylogenetic comparative methods \citep[e.g.][]{Felsenstein1985, Felsenstein1988, HarveyPagel1991, Pagel1993, Diaz1996, Price1997, GarlandIves2000, Rohlf2006, Freckleton2009, Hansen2012}, there have been few attempts to formally evaluate this --- in particular, for models more complex than a single rate BM model

\citet{Boettiger2012} developed a ``phylogenetic monte carlo'' approach to assess when a given comparative data set contained enough information to distinguish between two candidate models. In brief, their procedure was as follows: 1) select 2 candidate models $\mathcal{M}_0$ and $\mathcal{M}_1$; 2) fit $\mathcal{M}_0$ and $\mathcal{M}_1$ to the data using Maximum likelihood; 3) use the MLE for model parameters $\hat{\Theta}$ to simulate $n$ data sets; 4) for each of the $n$ data sets simulated under $\mathcal{M}_0$, fit both $\mathcal{M}_0$ and $\mathcal{M}_1$ and calculate difference in likelihood values $\delta = -2(L_{\mathcal{M}_0} - L_{\mathcal{M}_1})$; and 5) compare distribution of $\delta$ for datasets simulated under $\mathcal{M}_0$ with distribution of $\delta$ for datasets simulated under $\mathcal{M}_1$. \citet{Boettiger2012} demonstrated that their approach had much better ``classical'' statistical properties (specifically Type--I and Type--II errors) compared to using Information Theoretic methods of model selection, such as AIC \citep{Akaike1973}, AICc \citep{AICC}, and BIC \citep{Schwarz1978}. (Though we note that this is not an entirely fair comparison; Information Theoretic approaches differ philosophically from frequentist approaches to model selection [\citealt{BA2004}] and do not really have Type--I and Type--II error rates in the same way likelihood ratio tests [\citealt{Wilks1938}] are expected to have.)

\citet{SlaterPennell} also focused on the case of comparing two--models. Their method differs from that of \citet{Boettiger2012} in that theirs was a fully Bayesian posterior predictive approach specifically aimed at detecting ``early bursts'' of trait evolution \citep[\textit{sensu}][]{Simpson1944, Simpson1953, Harmon2010}. They sampled from the joint posterior distribution of the parameters, simulated data under the sampled parameters and then used two alternative summary statistics --- the relationship between the logarithm of the phylogenetic independent contrasts and the height above the root that the contrasts was inferred \citep[a.k.a. the ``node height test'';][]{FreckletonHarvey2006}, and the Mean Disparity Index \citep[MDI;][]{Harmon2003, Slater2010} --- to evaluate...

However, both of these cases used a simulation based approach to assess whether the data was informative enough to select amongs two candidate models. A broader question is whether a given model is a good fit to the data on its own terms --- compared to the universe of possible models we could consider.

The aim of our paper is two--fold. First, we wanted to address a fundamental macroevolutionary question: what are the major patterns in the evolution of plant functional traits? We focused on three important ecological traits --- seed size, specific leaf area and leaf nitrogen content --- which together encompass some of the major axes of life history and functional variation in vascular plants \citep{Cornwell2013}. \citep{Wright2004}
%% A lot more about the importance of these traits here
%% Will, I could really use your help on this section.

Investigating this empirical question forced us to address more theoretical ones, relating to the second aim of this paper: do our evolutionary models capture meaningful patterns when applied at this scale and how are we to know?

In this paper, we develop a general approach to assessing the adequacy of trait evolutionary models for continuous characters.

\section{Methods and data}

\subsection{Methodology}

Our approach is based on the use of Felsenstein's \citeyear{Felsenstein1973, Felsenstein1985} Phylogenetic Independent Contrasts (PIC) method, which we will briefly review \citep[for more details, see][]{Rohlf2001, Blomberg2012}. We have observed trait values $X_1, X_2, \ldots, X_N$ at the tips of a phylogenetic tree $\mathcal{T}$ consisting of $n$ species. Due to shared history of ancestry between the tips, $X_1, X_2, \ldots, X_n$ are not independent observations. To deal with this problem, Felsenstein suggested taking $n-1$ contrasts $c_1, c_2, \ldots, c_{n-1}$, the differences $X_{i} - X_{j}$ between the observations at tips $i$ and $j$. If we assume a BM model of trait evolution, in which variation accumulates directly proportional to time, these contrasts will be Independent and Identically Distributed (I.I.D.), hence the name PICs. The procedure can be described algorithmically. 1) Take the contrast $c = X_i - X_j$ at node $k$, where $k$ is the most recent common ancestor of tips $i$ and $j$. 2) Standardize the contrast by its dividing by its standard deviation, which under BM is $\sqrt{v_i + v_j}$, the square root of the sum of the branch lengths leading to $i$ and $j$, 3) Estimate a trait value for the ancestral node $k$ by taking the mean of its descendants' trait values, weighted by their branch lengths

\begin{equation}
X_k = \frac{(1 / v_i)X_i + (1 / v_j)X_j}{1/v_i + 1/v_j}.
\end{equation}

4) Lengthen the branch below $k$ by $v_i v_j / (v_i + v_j)$, in order to account for error in the estimation of $k$. Iterating across all nodes in the phylogeny, the result is a set of contrasts $\mathbf{c}$, which, as stated above, will be I.I.D., \textit{only if the true model which generated the observations was BM} \citep{Rohlf2001}. As our method described in this paper essentially evaluates whether this condition holds, we will refer to $\mathbf{c}$ as contrasts, rather than PICs throughout. 

The basic principle of our approach is straightforward. Given a set of $n$ observations, we can compute the $n-1$ contrasts. We calculate a set of summary statistics on the ``observed'' contrasts $\mathcal{S}^*$. We then simulate $N$ datasets using a Brownian motion model of trait evolution, compute the contrasts on each data set and calculate the same summary statistics $\mathcal{S}$. We then compare $\mathcal{S}^*$ to $\mathcal{S}$. If $\mathcal{S}^*$ lies in the tails of $\mathcal{S}$, our model is not capturing variation along a relevant axis. 

\subsubsection{Summary statistics}

To assess model adequacy we have chosen 6 summary statistics $\mathcal{S} = \lbrace \mathcal{S}_1, \ldots, \mathcal{S}_6 \rbrace$ 

\begin{enumerate}
\item[$\mathcal{S}_1$] $\overline{\mathbf{c}^2}$: the mean of the squared contrasts. This is equivalent to the Restricted maximum likelihood (REML) estimate of the Brownian motion rate parameter $\sigma^2$ \citep{Garland1992, Rohlf2001}. We chose this statistic to capture variation in the rate of trait evolution.

%% Check to make sure that the standard deviation is correct
\item[$\mathcal{S}_2$] $D_c$: the D--statistic obtained from Kolmolgorov-Smirnoff [SPELLING] test \citep{ks} from comparing the distribution of contrasts to that of a normal distribution with mean 0 and standard deviation $\sqrt{\overline{\mathbf{c}^2}}$. This is the expected distribution of the contrasts under BM \citep{Felsenstein1985, Rohlf2001}. We chose this to capture deviations from normality, such as would be produced if traits evolved via a ``jump diffusion'' type model \citep{Landis2013, Eastmanjump}, in which trait evolution may occasionally occur at rates much greater than the background rates \citep[see][]{PennellPE}.

\item[$\mathcal{S}_3$] $\mathrm{Var}[| \mathbf{c} |]$: the variance in the absolute value of the contrasts. This was chosen to capture heterogeneity in the rate of trait evolution.

%% I have thrown in the Garland et al. 1992
\item[$\mathcal{S}_4$] $m_{cv}$: the slope resulting from fitting a linear model between the absolute value of the contrasts and their expected variances. Each contrast has an expected variance equal to XX \citep{Felsenstein1985}. Under a model of BM, we expect no relationship between these. In using this, we are asking whether the contrasts are larger or smaller than we expect based on their branch lengths. If, for example, more evolution occured per unit time on short branches than long branches, we would observe a negative slope. \citep{Garlandetal1992}

\item[$\mathcal{S}_5$] $m_{ca}$: the slope resulting from fitting a linear model between the absolute value of the PICs and the inferred ancestral state. We estimated the ancestral state using the least--squared method suggested by \citep{Felsenstein1985} as this uses an identical procedure to that done when estimating PICs. This statistic will allow us to evaluate whether there is variation in rates relative to the trait value (e.g. do larger organisms evolve faster?) \citep{Garlandetal1992}

\item[$\mathcal{S}_6$] $m_{ct}$: the slope resulting from fitting a linear model between the absolute value of the contrasts and the height above the root at which they are inferred. This is alternatively known as the node height test \citep{FreckletonHarvey2006, SlaterPennell} for detecting early bursts of trait evolution and has been been previously used to assess the fit of BM models. \citep{Garlandetal1992}
\end{enumerate}

We have chosen these statistics because they capture a range of dimensions of variability wherein the model may not be adequate. However, they are certainly not exhaustive. One could, for instance, calculate the median of the squared contrasts, the skew of the distribution of contrasts, etc. If the generating (i.e. ``true'') model was known, we could use established procedures for selecting a set of sufficient (or approximately sufficient) summary statistics \citep[e.g.][]{MajoramJoyce, Wegmann2010}. However, the aim of our project is to seek a set of summary statistics that capture violations of the assumption that the contrasts are I.I.D. and thus we do not have the generating model in hand. Our chosen statistics appear to capture a wide variety of model misspecifications (see below) but this does not mean that they will necessarily capture \textit{any} model misspecification. Researchers interested in specific questions are encouraged to try alternate sets of summary statistics; we have made the software implementation of our approach as flexible as possible to accomodate alternative sets of statistics. (We describe how users of our software can easily implement these in the Supplementary Materials.)

\subsubsection{Rescaling the phylogeny}

While the above summary statistics are appropriate for a BM model of trait evolution, the same will not be true of alternative models. That is because under alternative models, we no longer expect the contrasts to have I.I.D. properties. Our solution to the problem is to use the estimated parameters of a more complex model $\Theta$ to create what we term a ``unit tree''. A unit tree is defined as a tree in which if the model we fit is the generating model, the data at the tips will be distributed as it would be under a BM process with a rate $\sigma^2$ equal to 1. (We note here that this is not technically a rate in the strict mathematical sense of the term \citep{Bookstein1987} but this is commonly used shorthand.)  Below, we define this concept more formally. Any phylogenetic tree $\mathcal{T}$ can be completely described by a $n \times n$ variance--covariance (vcv) matrix $\mathbf{C}$, where $n$ is equal to the number of tips in $\mathcal{T}$. The elements $C_{i,j}$ are the shared path--length from the root to the most recent common ancestor of $i$ and $j$ \citep{Piazza1975}. The diagonal elements ($i = j$) are simply the total distance from the root to the tips. For any model, we can describe a second matrix $\mathbf{\Sigma}$, which is the expected vcv matrix between observations at the tips. For example, under BM, in which variation accumulates proportionally to time under a single rate $\sigma^2$, 
\begin{equation}
\Sigma_{ij} = \sigma^2 C_{ij}
\end{equation}
and thus
\begin{equation}
\mathbf{\Sigma} = \sigma^2 \mathbf{C}.
\end{equation}
We can similarily construct $\mathbf{\Sigma}$ based on whatever model we are interested in. A vcv matrix $\mathbf{U}$, which describes the unit tree is then by definition equivalent to $\mathbf{\Sigma}$ for any model. We note that in practice we are using the estimated parameters $\hat{\Theta}$ from fitting the model such that there is some uncertainty in $\mathbf{\Sigma}$ and we will therefore refer to it as $\hat{\mathbf{\Sigma}}$ throughout \citep[see][]{Rohlf2001, Blomberg2012}. For the case of using a model with a single mean, we can transform $\mathbf{C}$ to $\mathbf{U}$ by... If we use a model with multiple means, such as a multi--optimum OU model \citep{ButlerKing2004, Beaulieu2012}, we must 

The approach we outlined above can be applied to any trait evolutionary model in which the observations at the tips are assumed to come from a multivariate normal distribution --- which applies to almost all continuous trait models currently used \citep{Omeara2012}. This includes single-- and multi--rate BM, OU with a single or multiple $\theta$ and $\alpha$ parameters as well as variations thereof \citep[see][for an example of an evolutionary model for which this does not hold]{Landis2012}.

Additionally, this approach can is applicable to ``phylogenetic regression'' (e.g. ``phylogenetic generalized least squares''; PGLS) analyses, in which the aim is to test for evolutionary correlations between traits \citep{Grafen1989, Rohlf2001}. In general terms, we can write down the regression equation as:

\begin{equation}
\mathbf{y} = \mathbf{b}\mathbf{X} + \epsilon .
\end{equation} 

Here $\mathbf{y}$ is the $n \times 1$ vector of the dependent variable, $\mathbf{b}$ is the vector of partial regression coefficients, $\mathbf{X}$ is the $n \times q$ matrix of independent variables ($q$ being the number of traits included in the model) and $\epsilon$ is the error term. In ordinary least squared (OLS) regression, we assume that the error term $\epsilon$ is normally distributed with parameters $(0, \sigma^2)$. However, when we taken into account the covariation between tip observation due to the pattern of shared ancestry, the residuals are no longer I.I.D. and will be distributed $\mathcal{N} (0, \hat{\mathbf{\Sigma}})$, where again $\hat{\mathbf{\Sigma}}$ is the estimated vcv matrix from fitting the evolutinary model. While the original version of PGLS assumed a BM model of trait evolution, this has been expanded to include a number of models, including the $\lambda$ model \citep{Pagel1997, Freckleton2009, Revell2010} and OU \citep{Hansen2008, Labra2009, Bartoszek2012} among others \citep[see also][]{Lynch1991, Hadfield2010}. Similar to the univariate case described above, these regression equations fundamentally depend on the use of an appropriate model \citep{Hansen2012SysBio}. To assess the adequacy of the model used to estimate $\hat{\mathbf{\Sigma}}$, we can use an identical approach to the one described above, but rather that take the contrasts of the trait values, we take the contrasts of the residuals from fitting our regression model to the data. This may not appear intuitive, but phylogenetic regression models do not apply the covariance matrix to the data themselves, only to the residuals; in other words, the phylogenetic structure of actual data does not matter, only that of the residuals \citep{Rohlf2001}. Therefore, if the model specifiying $\mathbf{\Sigma}$ is correct, the contrasts of the residuals taken along the unit tree will also be I.I.D.

We note that while the approach presented here can be used to evaluate the adequacy of the trait model, it does not assess the adequacy of the linear component of the model $\mathbf{y} = \mathbf{b}\mathbf{X}$. It is also worth noting here that even if $\hat{\mathbf{\Sigma}}$ is misspecified, such as by using an incorrect model or an incorrect tree, the GLS estimate of the slope $\mathbf{b}$ will be unbiased \citep{Rao1999}; however the variance of the estimator will be too small \citep{Rohlf2006}. And where deviations from the true $\mathbf{b}$ do occur, the directions of the deviations (i.e. a higher or lower regression coefficient) are not necessarily predictable \citep{Rohlf2006}. 

\subsubsection{Parametric bootstrapping and posterior predictive simulations}

There are two alternative ways to employ this method. The first is parametric bootstrapping in which the maximum likelihood values of a set of model parameters $\hat{\Theta}$ are estimated. $\hat{\Theta}$ is used to estimate $\hat{\mathbf{\Sigma}}$ and generate the unit tree. We calculate our set of summary statistics on our observed data $\mathcal{S}^*$. We then simulate \textit{n} datasets under a BM process with $\sigma^2 = 1$. On each simulated dataset we calculate our set of summary statistics $\mathcal{S}$. We then compare $\mathcal{S}^*$ to the distribution of $\mathcal{S}$. If any of our observed statistics lies in the tails of that of the simulated data, we conclude that our model is not accounting for variation along the axis captured by the summary statistic. 

\subsubsection{Simulations}


\subsection{Empirical analyses}

\subsubsection{Phylogenetic tree and Trait Data}

We used a phylogeny of Angiosperms from a recent study by \citet{ZanneBigTree}. The tree contains 31,749 land plant taxa and covers xx \% of familial diversity and yy \% of generic diversity across all Angiosperms. We will not provide the full details on the phylogeny here and refer interested readers to the original publication \citep{ZanneBigTree}. But in brief, the data matrix was constructed using 7 loci (\textit{18S rDNA}, \textit{26S rDNA}, \textit{ITS}, \textit{matK}, \textit{rbcL}, \textit{atpB}, and \textit{trnLF}) accessed from \textsc{GenBank}, using the pipeline \texttt{PHLAWD} \citep{phlawd}. The phylogeny was built using \texttt{RAxML} \citep{raxml} and time--calibrated using Congruification \citep{Eastmancongruify} with divergence times estimates from Tank, Eastman, Pennell, P. Soltis, D. Soltis and Harmon (\textit{unpublished}) and the Penalized Likelihood method \citep{Sanderson1997, treepl} for estimating branch lengths.

For the purposes of this study, we used the MLE point estimate of the phylogeny. We could have used a set of bootstrapped trees and ran our analyses across all of them, but for the analyses we have conducted, it should not qualitatively affect our results. This tree is published on \textsc{treebase} (assession number) and \textsc{dryad} (accession number).

\subsubsection{Adequacy of models for plant functional traits}


\section{Results}

\subsection{Simulation results}

\subsection{Adequacy of models for plant functional traits}


\section{Discussion}

The method we have described here can apply to any model of trait evolution which can be described by a multivariate normal distribution. That is, where the original tree can be transformed from the model fit to a unit tree. Our approach can therefore not be applied to other types of models, such as those which assume a long--tailed distribution of trait change \citep[e.g.][]{Landis2012} or state--dependent diversification models \citep[e.g.][]{Bokma2008, Bokma2010, FitzJohn2010}. And it is not applicable to models of discrete character evolution, such as the Mk model \citep{Pagel1994, Lewis2001} and variations thereof \citep[e.g.][]{Maddison2007, WagnerMarcot2010, Mazeralli2012, Beaulieu2013} \citep[but see Felsenstein's threshold model;][]{Felsenstein2005, Felsenstein2012}. Methods to evaluate the adequacy of discrete character models are also sorely needed, though the statistics involved will necessarily be different.
%% Here I should find some way to cite Read and Nee 1995.
Another complication which has the potential to mislead inferences is based sampling of traits \citep{Freckletoninaction, longeuroname, FitzJohnwoody}. All the trait models considered in this study assume that evolution has occured independently on each branch, such that as long as the taxa included in the phylogeny are a more--or--less random sample, the parameter estimates should not be biased as a result \citep{PennellHarmon}. However, in many cases \citep[especially when using large trait databases][]{FitzJohnwoody} this will not be the case. Importantly this will not be detected by the model adequacy approach presented in this paper: the bias in the parameter estimates will be systematically the same across all the simulated datasets.

While we describe this as a novel approach much of it steams directly from earlier work in the field. In the 1980s and 1990s --- the pioneering days of comparative biology --- much discussion surrounded the appropriateness of various methods and models.
\section{Concluding remarks}




\section{A note on implementation}

The method described here has been implemented in the R statistical environment \citep{R}. It is now in the \texttt{GEIGER} package \citep{geiger}, available on CRAN. For this project, we have also adopted code from the \texttt{ape} \citep{ape}, \texttt{diversitree} \citep{FitzJohn2012}, \texttt{multicore} \citep{multicore} and \texttt{ggplot2} \citep{ggplot2} repositories. We have written functions to parse the output of a number of different programs for fitting trait evolution models (see \ref{supp-table-fxns}). As this is approach was developed to be general, we have written the code in such a way that users can include their own summary statistics and trait models in the analyses; we include demonstrations of how this can be done in the supplementary material. All source code for this project at \texttt{www.github.com/mwpennell/modeladequacy} and is soon to be implemented in the comparative methods workflow software \texttt{Arbor} \citep{HarmonArbor}. 

\section{Acknowledgments}

We would like to thank the members of the Tempo and Mode of Trait Evolution Working Group at the National Evolutionary Synthesis Center (NESCent) as well as NESCent for funding our group. We thank Jon Eastman for his assistance with the MCMC algorithm for fitting trait evolutionary models and Rich FitzJohn, Josef Uyeda and Jeremy Beaulieu for their insightful comments on this project.



\newpage
\bibliographystyle{sysbio}
\bibliography{model_ad.bib}

\end{document}