\documentclass[12pt]{article}
\usepackage[osf]{mathpazo}
\usepackage{ms}
\usepackage{natbib}
\usepackage{lineno}
\usepackage{graphicx}
\usepackage{caption}
\modulolinenumbers[5]
\linenumbers

\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\title{The adequacy of trait models and the rise of angiosperm functional diversity}
\author{
Matthew W. Pennell$^{1}$, William K. Cornwell$^{2}$,\\
Richard G. FitzJohn$^{3}$ \& Luke J. Harmon$^{1}$
}

\date{}
\affiliation{
\begin{center}
\textit{
$^{1}$ Department of Biological Sciences \& Institute for Bioinformatics and Evolutionary Studies, University of Idaho, Moscow, ID 83844, U.S.A.\\[0.5cm]
$^{2}$ School of BEES, University of New South Wales, Sydney, NSW 2052, Australia\\[0.5cm]
$^{3}$ Department of Biological Sciences, Macquarie University, Sydney, NSW 2109, Australia\\[0.5cm]
Email for correspondence: mwpennell@gmail.com}
\end{center}
}
\runninghead{are macroevolutionary models adequate?}


\begin{document}
\mstitlepage
\parindent=1.5em
\addtolength{\parskip}{.3em}

\begin{abstract}
Investigating evolutionary hypotheses using interspecific data necessitates the use of a statistical model which describes the pattern of trait evolution along a phylogeny. A wide variety of such models have been proposed and fitting and comparing these is now a routine component of comparative studies. However, an often over--looked statistical question is whether the models we are using are actually capturing the relevant variation in traits. A model may be the best fit (using some model selection criterion) relative to a set of candidate models but it may still be a poor fit in an absolute sense. As the scale of comparative datasets and questions continues to increase, the question of model fit becomes even more important to consider. Here we develop a general framework for assessing the fit (or, adequacy) of quantitative trait models, based on Felsenstein's Independent Contrasts method. Our framework can be applied to arbitrarily complex trait models for which the tip values are assumed to multivariate normally distributed. We then use this approach to assess whether simple models are reasonabe descriptors of the macroevolutionary dynamics of three important angiosperm functional traits: seed size, specific leaf area, and nitrogen content. We analyzed 1xx comparative datasets across the angiosperm phylogeny and demonstrate that there is tremendous variability in the adequacy of simple models across groups, traits and time--scales. We argue that the assessment of model adequacy should be included in all comparative studies and that the framework we present can be generally applied to this end.  
\end{abstract}

\vfill

\newpage


\begin{quotation}
\noindent There are known knowns; there are things we know we don't know. There are known unknowns; that is to say, there are things that we now know we don't know. But there are also unknown unknowns --- there are things we do not know we don't know. 

---Fmr. U.S. Secretary of Defense Donald Rumsfeld
\end{quotation}


\noident Angiosperms are one of the most spectacular radiations in the history of the earth. The $>$300,000 species which make up the clade have diversified into a huge array of functional forms, from the grasses of the Serengeti to the trees of the Amazonian rainforest, and have come to dominate most terrestrial ecosystems. As researchers interested in macroevolution, we would like to understand how such trait diversity came to be --- the who, what, when, where and whys of deep time. For example, seed size is associated with many other life--history triats [give example] and varies $\sim$ 11 orders of magnitude across extant angiosperm species \citep{Westoby1992TREE, Moles2005}. [Talk about other things we know about seed size; cite Lord 1995, Westoby 2002, Moles 2005, Cornwell 2013, etc.]

There are a number of questions we might want to ask about the evolution of seed size. For example, we may want to know whether the diversity in seed size tends to arise early or late in the history of a clade. [What did Stebbins say about this? Other previous workers/hypotheses?] Alternatively, we may want to investigate whether the evolution of seed size was associated with the evolution of other traits, including other life--history traits or genomic characteristics. Our ability to ask such questions at macroevolutionary scales requires that the diversity be placed in a phylogenetic context and that we have a statistical model which describes the pattern of evolution across the group. 

%%Might want to use subheadings here
%%e.g. "Current practice in comparative methods" or something
The last few decades have seen a tremendous growth in statistical methods that make use of phylogenetic trees and comparative data to address macroevolutionary questions \citep[for recent reviews, see][]{Omeara2012, PennellHarmon}. Despite the diversity of approaches, most modern applications of phylogenetic comparative methods involve the same general procedure: 1) come up with set of statistical models which a researcher thinks may characterize the pattern of evolution; 2) fit these models to the data using maximum likelihood, Bayesian inference or some other technique (such as least squares or Restricted maximum likelihood, etc.); 3) compare the relative fits of the model using some model selection criteria; 4) use the best--fitting model to make evolutionary inferences. This last step may involve either using the model directly to make inferences regarding the tempo and mode of evolution \citep[e.g.][]{HansenMartins1996, Mooers1999, Harmon2010, Hunt2012} or else to use the fitted model to make secondary inferences, such as that of a correlation between traits \citep[e.g.][]{Felsenstein1985, Grafen1989, Freckleton2009} or to ``reconstruct'' the evolutionary history of a character \citep[e.g.][]{Schluter1997, Huelsenbeck2003}. 

A variety of models have been proposed for the evolution of quantitative traits, such as seed size, plant height, etc. (In this paper, we focus on these types of traits, see \textit{Discussion} for comments on other varieties). Brownian motion (BM), in which evolution proceeds via an undirected random walk such that the variance in the trait value accumulates proportional to time, is the oldest, and most commonly applied model of continuous character evolution. Originally developed as a model for estimating phylogenetic trees from allelic frequencies \citep{EC1964}, BM has been used as a more general model of phylogenetic change in phylogenetics \citep{Felsenstein1973, Thompson1975, Felsenstein1988} and is the underlying model in Felsenstein's phylogenetically independent contrasts \citep[PICs;][]{Felsenstein1985} for investigating correlated evolution. Another model is the Ornstein--Uhlenbeck process \citep[OU;][]{Hansen1997} which can be loosely described as ``evolution on a spring'' --- variance accumulates at a rate $\sigma^2$, but is ``pulled'' towards a mean value $\theta$ with some strength $\alpha$. Alternate models that have been considered include: the ``Early Burst'' \citep[EB;][]{Blomberg2003, Harmon2010, SlaterPennell} model, in which most of the evolution occurs early in the clades history; a BM model with a trend in the mean trait value \citep{Hunt2006}; models depicting jumps in phenotypic space \citep{Landis2012, Eastmanlevy}; and other variations on the models described above \citep[e.g.][]{Pagel1997, Pagel1999, ButlerKing2004, Omeara2006, Eastman2011, Beaulieu2012, SlaterMEE}.

We can fit any of these models to comparative data but after estimating parameters, there are a number of important questions which remain unanswered. One is that of interpretation --- what (if anything) does a given model tell us about the process of evolution \citep{HansenMartins1996, Hansen2012, PennellHarmon}? Another is statistical --- is the model we are using capturing a sufficient amount of the variation in the data to make inferences regarding the question of interest? The latter question, known in statistical teminology as model adequacy, is the subject of our investigation here.

In many statistical applications, assessing the absolute fit of the model \textit{a posteriori} is a routine procedure \citep{Gelmanbook}. Before drawing inferences about the parameters of the model, we want to know whether the model we used is adequately capturing the relevant patterns in the data. In phylogenetics, model adequacy has been investigated for models of sequence evolution \citep[e.g.][]{GautLewis1995, SullivanSwofford, Goldman, HuelsenbeckBull1996, SandersonKim, Bollback2002, Ripplinger2010, Lewis2013, Brown2013} and forms the basis for the Decision--Theoretic approach to model selection \citep{Minin2003, Abdo2005, SullivanJoyce2005}. In contrast, in the comparative biology, while there has been a great deal of discussion of the appropriateness of various models \citep[e.g.][]{Felsenstein1985, Felsenstein1988, HarveyPagel1991, Pagel1993, Diaz1996, Price1997, GarlandIves2000, Rohlf2006, Freckleton2009, Hansen2012}, there have been few attempts to formally evaluate this --- in particular, for models more complex than a single rate BM model. Developing approaches to assess model adequacy is all the more pertinent as the scale of comparative data continues to grow. Comparative analyses are increasingly being performed on very large datasets including thousands and even tens of thousands of species \citep[e.g.][]{Coopermammal, Jetz2012, Rabosky2013, Cornwell2013, PyronBubrink2013, Zanne2013}. While a simple model of trait evolution may suffice for a small clade of closely related organisms (e.g. seed size across Oak species), it seems unlikely that this would be true at much broader scales (e.g. seed size across all Angiosperms).

Recently, two papers have developed simulation--based approaches to assessing the fit of models. \citet{Boettiger2012} developed a ``phylogenetic monte carlo'' approach to assess when a given comparative data set contained enough information to distinguish between two candidate models. In brief, their procedure was as follows: 1) select 2 candidate models $\mathcal{M}_0$ and $\mathcal{M}_1$; 2) fit $\mathcal{M}_0$ and $\mathcal{M}_1$ to the data using Maximum likelihood; 3) use the MLE for model parameters $\hat{\Theta}$ to simulate $n$ data sets; 4) for each of the $n$ data sets simulated under $\mathcal{M}_0$, fit both $\mathcal{M}_0$ and $\mathcal{M}_1$ and calculate difference in likelihood values $\delta = -2(L_{\mathcal{M}_0} - L_{\mathcal{M}_1})$; and 5) compare distribution of $\delta$ for datasets simulated under $\mathcal{M}_0$ with distribution of $\delta$ for datasets simulated under $\mathcal{M}_1$. \citet{Boettiger2012} demonstrated that their approach had much better ``classical'' statistical properties (specifically Type--I and Type--II errors) compared to using Information Theoretic methods of model selection, such as AIC \citep{Akaike1973}, AICc \citep{AICC}, and BIC \citep{Schwarz1978}. (Though we note that this is not an entirely fair comparison; Information Theoretic approaches differ philosophically from frequentist approaches to model selection [\citealt{BA2004}] and do not really have Type--I and Type--II error rates in the same way likelihood ratio tests [\citealt{Wilks1938}] are expected to have.)

In the other paper, \citet{SlaterPennell} also focused on the case of comparing two--models. Their method differs from that of \citet{Boettiger2012} in that theirs was a fully Bayesian posterior predictive approach specifically aimed at detecting ``early bursts'' of trait evolution \citep[\textit{sensu}][]{Simpson1944, Simpson1953, Harmon2010}. They sampled from the joint posterior distribution of the parameters, simulated data under the sampled parameters and then used two alternative summary statistics --- the relationship between the logarithm of the phylogenetic independent contrasts and the height above the root that the contrasts was inferred \citep[a.k.a. the ``node height test'';][]{FreckletonHarvey2006}, and the Mean Disparity Index \citep[MDI;][]{Harmon2003, Slater2010} --- to evaluate...

However, both of these cases used a simulation based approach to assess whether the data was informative enough to select amongst two candidate models. A broader question is whether a given model is a good fit to the data on its own terms --- that is, compared to the universe of possible models we could consider. Here we develop a approach to statistically evaluate for models of continuous character evolution. The basic idea of our approach is as follows. If we have a trait that has evolved on the phylogeny via a BM process with a rate (usually denoted $\sigma^2$) equal to 1, the contrasts \citep[\textit{sensu}][see below]{Felsenstein1985} will be independent and identically distributed (I.I.D.) according to a standard normal distribution $\mathcal{N}(0,1)$. In our approach, we first rescale the phylogeny according to the model fit, such that if the model were the generating model, the trait data would be distributed as it would be under a BM process with a rate of 1 (which we will refer to as a ``unit tree''). We then calculate the contrasts at each node and evaluate whether they are indeed distributed as we expect using a variety of summary statistics. We then simulate datasets under BM (again, using $\sigma^2 = 1$) and calculate our summary statistics on each simulated dataset. We then compare our observed to our simulated summary statistics (details below). 

We then use our approach to evaluate whether simple models are adequate for the evolution of a number of plant functional traits and we do at a number of different scales. Specifically we sought to address...
%% A lot more about the importance of these traits here
%% Will, I could really use your help on this section.


\section{Methods and data}

\subsection{Methodology}

Our approach is based on the use of Felsenstein's \citealtyear{Felsenstein1973, Felsenstein1985} Phylogenetic Independent Contrasts (PIC) method, which we will briefly review \citep[for more details, see][]{Felsenstein1985, Rohlf2001, Blomberg2012}. We have observed trait values $X_1, X_2, \ldots, X_n$ at the tips of a phylogenetic tree $\mathcal{T}$ consisting of $n$ species. Due to shared history of ancestry between the tips, $X_1, X_2, \ldots, X_n$ are not independent observations. To deal with this problem, Felsenstein suggested taking $n-1$ contrasts $c_1, c_2, \ldots, c_{n-1}$, the differences $X_{i} - X_{j}$ between the observations at tips $i$ and $j$. If we assume a BM model of trait evolution, in which variation accumulates directly proportional to time, these contrasts will be independent and identically distributed (I.I.D.), hence the name PICs. The procedure can be described algorithmically: 1) Take the contrast $c = X_i - X_j$ at node $k$, where $k$ is the most recent common ancestor of tips $i$ and $j$. 2) Standardize the contrast by its dividing by its standard deviation, which under BM is $\sqrt{v_i + v_j}$, the square root of the sum of the branch lengths leading to $i$ and $j$, 3) Estimate a trait value for the ancestral node $k$ by taking the mean of its descendants' trait values, weighted by their branch lengths

\begin{equation}
X_k = \frac{(1 / v_i)X_i + (1 / v_j)X_j}{1/v_i + 1/v_j}.
\end{equation}

4) Lengthen the branch below $k$ by $v_i v_j / (v_i + v_j)$, in order to account for error in the estimation of $k$. Iterating across all nodes in the phylogeny, the result is a set of contrasts $\mathbf{c}$, which, as stated above, will be I.I.D. \textit{only if the model which generated the observations was BM} \citep{Rohlf2001}. As our method described in this paper essentially evaluates whether this condition holds, we will refer to $\mathbf{c}$ as contrasts, rather than PICs throughout. 

The basic principle of our approach is straightforward. Given a set of $n$ observations, we can compute the $n-1$ contrasts. We calculate a set of summary statistics on the ``observed'' contrasts $\mathcal{S}^*$. We then simulate $N$ datasets using a Brownian motion model of trait evolution, compute the contrasts on each data set and calculate the same summary statistics $\mathcal{S}$. We then compare $\mathcal{S}^*$ to $\mathcal{S}$. If $\mathcal{S}^*$ lies in the tails of $\mathcal{S}$, our model is not capturing variation along a relevant axis. 

\subsubsection{Summary statistics}

To assess model adequacy we have chosen 6 summary statistics $\mathcal{S} = \lbrace \mathcal{S}_1, \ldots, \mathcal{S}_6 \rbrace$ 

\begin{enumerate}
\item[$\mathcal{S}_1$] $\overline{\mathbf{c}^2}$: the mean of the squared contrasts. This is equivalent to the Restricted maximum likelihood (REML) estimate of the Brownian motion rate parameter $\sigma^2$ \citep{Garland1992, Rohlf2001}. We chose this statistic to capture variation in the rate of trait evolution.

%% Check to make sure that the standard deviation is correct
%% Why not use 1 as a standard deviation?
%% I suggest that the answer to this question is that we are already testing for
%% deviations in rate. Using the estimated sigsq will make this test more independent
%% and capture what we really want -- deviations from normality
\item[$\mathcal{S}_2$] $D_c$: the D--statistic obtained from Kolmolgorov-Smirnov test \citep{ks} from comparing the distribution of contrasts to that of a normal distribution with mean 0 and standard deviation $\sqrt{\overline{\mathbf{c}^2}}$ --- the expected distribution of the contrasts under BM \citep{Felsenstein1985, Rohlf2001}. We chose this to capture deviations from normality, such as would be produced if traits evolved via a ``jump diffusion'' type model \citep{Landis2013, Eastmanjump}, in which trait evolution may occasionally occur at rates much greater than the background rates \citep[see][]{PennellPE}.

\item[$\mathcal{S}_3$] $\mathrm{Var}[| \mathbf{c} |]$: the variance in the absolute value of the contrasts. This was chosen to capture heterogeneity in the rate of trait evolution.

%% I have thrown in the Garland et al. 1992 citation as they use these stats but need to better integrate this into text
\item[$\mathcal{S}_4$] $m_{cv}$: the slope resulting from fitting a linear model between the absolute value of the contrasts and their expected variances. Each contrast has an expected variance equal to XX \citep{Felsenstein1985}. Under a model of BM, we expect no relationship between these. In using this, we are asking whether the contrasts are larger or smaller than we expect based on their branch lengths. If, for example, more evolution occured per unit time on short branches than long branches, we would observe a negative slope.

\item[$\mathcal{S}_5$] $m_{ca}$: the slope resulting from fitting a linear model between the absolute value of the PICs and the inferred ancestral state. We estimated the ancestral state using the least--squared method suggested by \citep{Felsenstein1985} as this uses an identical procedure to that done when estimating PICs. This statistic will allow us to evaluate whether there is variation in rates relative to the trait value (e.g. do larger organisms evolve faster?)

\item[$\mathcal{S}_6$] $m_{ct}$: the slope resulting from fitting a linear model between the absolute value of the contrasts and the height above the root at which they are inferred. This is alternatively known as the node height test \citep{FreckletonHarvey2006, SlaterPennell} for detecting early bursts of trait evolution and has been been previously used to assess the fit of BM models. 
\end{enumerate}

The three summary statistics fitting a linear model between the contrasts and some aspect of the data (i.e. $\mathcal{S}_4, \mathcal{S}_5, \mathcal{S}_6$) were developed by Garland and colleagues \citep{Garland1992, Garland1993,  Diaz1996} to test for adequacy of BM prior to performing PICs \citep{Felsenstein1985} or phylogenetic regression \citep{Grafen1989}. They were implemented in the formerly popular program PDAP \citep{GarlandPDAP, Midford2005} and have been well used in the comparative methods literature (at least until more recently).  

We have chosen these statistics because they capture a range of dimensions of variability wherein the model may not be adequate. However, they are certainly not exhaustive. One could, for instance, calculate the median of the squared contrasts, the skew of the distribution of contrasts, etc. If the generating (i.e. ``true'') model was known, we could use established procedures for selecting a set of sufficient (or approximately sufficient) summary statistics \citep[e.g.][]{MajoramJoyce, Wegmann2010}. However, the aim of our project is to seek a set of summary statistics that capture violations of the assumption that the contrasts are I.I.D. and thus we do not have the generating model in hand. Our chosen statistics appear to capture a wide variety of model misspecifications (see below) but this does not mean that they will necessarily capture \textit{any} model misspecification. Researchers interested in specific questions are encouraged to try alternate sets of summary statistics; we have made the software implementation of our approach as flexible as possible to accomodate alternative sets of statistics. (We describe how users of our software can easily implement these in the Supplementary Materials.) We also note that a model will be considered inadequate if it is rejected by \textit{any} of the summary statistics that can be used. In principle, one could evaluate the model fit with a large number of summary statistics. Even if the model is not rejected by any of the summary statistics used, we still do not know if there adding another statistic would cause us to reject the model. This is perhaps a somewhat unsatisfying situation to be in but we have to live with it. The choice of how many summary statistics to use is ultimately one of balancing statistical intuition and computational effort. 

\subsubsection{Rescaling the phylogeny}

While the above summary statistics are appropriate for a BM model of trait evolution, the same will not be true of alternative models. That is because under alternative models, we no longer expect the contrasts to have I.I.D. properties. Our solution to the problem is to use the estimated parameters of a more complex model $\Theta$ to create what we term a ``unit tree''. A unit tree is defined as a tree in which if the model we fit is the generating model, the data at the tips will be distributed as it would be under a BM process with a rate $\sigma^2$ equal to 1. Here, we define this concept more formally. Any phylogenetic tree $\mathcal{T}$ can be completely described by a $n \times n$ variance--covariance (vcv) matrix $\mathbf{C}$, where $n$ is equal to the number of tips in $\mathcal{T}$. The elements $C_{i,j}$ are the shared path--length from the root to the most recent common ancestor of $i$ and $j$ \citep{Piazza1975}. The diagonal elements ($i = j$) are simply the total distance from the root to the tips. For any model, we can describe a second matrix $\mathbf{\Sigma}$, which is the expected vcv matrix between observations at the tips. For example, under BM, in which variation accumulates proportionally to time under a single rate $\sigma^2$, 
\begin{equation}
\Sigma_{ij} = \sigma^2 C_{ij}
\end{equation}
and thus
\begin{equation}
\mathbf{\Sigma} = \sigma^2 \mathbf{C}.
\end{equation}
We can similarily construct $\mathbf{\Sigma}$ based on whatever model we are interested in. \citet{HoAne2013} describe a condition which they refer to as the ``3 point condition''. A vcv matrix $\mathbf{\Sigma}$ satisfies this condition if, and only if, it is symmetric, has only non--negative entries and that for any $i,j,k$, the two smallest of $\Sigma_{i,j}, \Sigma_{j,k}, \Sigma_{i,k}$ are equal. They prove that this a) this condition is only met for rooted phylogenies with trait data generated under BM; and b) that any $\mathbf{\Sigma}}$ based on a multivariate Gaussian distribution can be rescaled to satisfy the condition. Our concept of a unit tree is equivalent to their ``3 point condition'' which the additional requirement that rate parameter of the BM model is equal to 1. This last requirement puts all models on the same scale so that it is more intutive to compare them; however, this is not statistically necessary. As stated above, a unit tree can be constructed from any multivariate Gaussian model such as mulit--rate BM \citep{Omeara2006, Thomas2006, Eastman2011, Revell2012, Motmot}, Pagel's $\lambda, \kappa, and \delta$ transformations \citep{Pagel1997, Pagel1999}, the AC/DC model \citep{Blomberg2003} \citep[a.k.a. `EB';][]{Harmon2010, SlaterPennell}, models with a directional trend and others \citep[e.g.][]{SlaterMEE}. OU models can also be rescaled to form a unit tree; this is true of both multi--optimum \citep{ButlerKing2004} models and the more generalized form \citep{Beaulieu2012} in which the optimum, $\sigma^2$ and $\alpha$ values all vary across the tree \citep{HoAne2013}. We note that all of above applies to both ultrametric and non--ultrametric phylogenies \citep{HoAne2013}, such that the methods we describe can be applied to trees consisting of fossil taxa as well.

The approach we describe is applicable to both univariate and multivariate analyses, such as ``phylogenetic regression'' (e.g. ``phylogenetic generalized least squares''; PGLS) analyses, in which the aim is to test for evolutionary correlations between traits \citep{Grafen1989, Rohlf2001}. In general terms, we can write down the regression equation as:

\begin{equation}
\mathbf{y} = \mathbf{b}\mathbf{X} + \epsilon .
\end{equation} 

Here $\mathbf{y}$ is the $n \times 1$ vector of the dependent variable, $\mathbf{b}$ is the vector of partial regression coefficients, $\mathbf{X}$ is the $n \times q$ matrix of independent variables ($q$ being the number of traits included in the model) and $\epsilon$ is the error term. In ordinary least squared (OLS) regression, we assume that the error term $\epsilon$ is normally distributed with parameters $(0, \sigma^2)$. However, when we taken into account the covariation between tip observation due to the pattern of shared ancestry, the residuals are no longer I.I.D. and will be distributed $\mathcal{N} (0, \mathbf{\Sigma})$, where again $\mathbf{\Sigma}$ is the estimated vcv matrix from fitting the evolutinary model. While the original version of PGLS assumed a BM model of trait evolution, this has been expanded to include a number of models, including the $\lambda$ model \citep{Pagel1997, Freckleton2009, Revell2010} and OU \citep{Hansen2008, Labra2009, Bartoszek2012} among others \citep[see also][]{Lynch1991, Hadfield2010}. Similar to the univariate case described above, these regression equations fundamentally depend on the use of an appropriate model \citep{Hansen2012SysBio}. To assess the adequacy of the model used to estimate $\mathbf{\Sigma}$, we can use an identical approach to the one described above, but rather that take the contrasts of the trait values, we take the contrasts of the residuals from fitting our regression model to the data. This may not appear intuitive, but phylogenetic regression models do not apply the covariance matrix to the data themselves, only to the residuals; in other words, the phylogenetic structure of actual data does not matter, only that of the residuals \citep{Rohlf2001}. Therefore, if the model specifiying $\mathbf{\Sigma}$ is correct, the contrasts of the residuals taken along the unit tree will also be I.I.D.

We note that while the approach presented here can be used to evaluate the adequacy of the trait model, it does not assess the adequacy of the linear component of the model $\mathbf{y} = \mathbf{b}\mathbf{X}$. It is also worth noting here that even if $\hat{\mathbf{\Sigma}}$ is misspecified, such as by using an incorrect model or an incorrect tree, the GLS estimate of the slope $\mathbf{b}$ will be unbiased \citep{Rao1999}; however the variance of the estimator will be too small \citep{Rohlf2006}. And where deviations from the true $\mathbf{b}$ do occur, the directions of the deviations (i.e. a higher or lower regression coefficient) are not necessarily predictable \citep{Rohlf2006}. 

\subsubsection{Posterior predictive simulations versus parametric bootstrapping}
%% Need to make the two paragraphs more similar
%% Not sure how formal we need to be here!!
There are two alternative ways to conceive of our method. From a Bayesian perspective, our approach can be considered as a form of posterior predictive check \citep{Rubin1984, Gelman1996, Gelman2003}. In this case, we fit a model using Bayesian machinery (e.g. Markov chain Monte Carlo, MCMC, methods), then sample parameter values from the joint posterior distribution. For each sampled parameter set, we rescale the original phylogeny to a unit tree, calculate a set of summary statistics and then simulated a dataset on this rescaled tree. We can then compare the distribution of summary statistics from the observed dataset to that of the simulated datasets using some classical statistical method. He we chose to use a Kolmorgorov-Smirnov (KS) test \citep{ks} as we are interested in deviations across the distribution and not just differences in the mean or variance.


Alternatively, one can employ our method in a frequentist/likelihood framework. As such, our method can be construed as a parametric bootstrapping approach \citep{Efronbootstrap}. First, the maximum likelihood (or alternatively Restricted maximum likelihood; REML) values of a set of model parameters $\hat{\Theta}$ are estimated. $\hat{\Theta}$ is used to estimate $\hat{\mathbf{\Sigma}}$ and generate the unit tree. We calculate our set of summary statistics on our observed data $\mathcal{S}^*$. We then simulate \textit{n} datasets under a BM process with $\sigma^2 = 1$. On each simulated dataset we calculate our set of summary statistics $\mathcal{S}$. We then compare $\mathcal{S}^*$ to the distribution of $\mathcal{S}$. If any of our observed statistics lies in the tails of that of the simulated data, we conclude that our model is not accounting for variation along the axis captured by the summary statistic. 



\subsubsection{Simulations}


\subsection{Empirical analyses}

\subsubsection{Phylogenetic tree and Trait Data}

We used a phylogeny of Angiosperms from a recent study by \citet{ZanneBigTree}. The tree contains 31,749 land plant taxa and covers xx \% of familial diversity and yy \% of generic diversity across all Angiosperms. We will not provide the full details on the phylogeny here and refer interested readers to the original publication \citep{ZanneBigTree}. But in brief, the data matrix was constructed using 7 loci (\textit{18S rDNA}, \textit{26S rDNA}, \textit{ITS}, \textit{matK}, \textit{rbcL}, \textit{atpB}, and \textit{trnLF}) accessed from \textsc{GenBank}, using the pipeline \texttt{PHLAWD} \citep{phlawd}. The phylogeny was built using \texttt{RAxML} \citep{raxml} and time--calibrated using Congruification \citep{Eastmancongruify} with divergence times estimates from Tank, Eastman, Pennell, P. Soltis, D. Soltis and Harmon (\textit{in prep.}) and the Penalized Likelihood method \citep{Sanderson1997, treepl} for estimating branch lengths.

For the purposes of this study, we used the MLE point estimate of the phylogeny. We could have used a set of bootstrapped trees and ran our analyses across all of them, but for the analyses we have conducted, it should not qualitatively affect our results. This tree is published in \textsc{treebase} (assession number) and \textsc{dryad} (accession number).

\subsubsection{The adequacy of models for functional trait evolution in Angiosperms}


\section{Results}

\subsection{Simulation results}

\subsection{Adequacy of models for plant functional traits}


\section{Discussion}

The method we have described here can apply to any model of trait evolution which can be described by a multivariate normal distribution. That is, where the original tree can be transformed from the model fit to a unit tree. Our approach can therefore not be applied to other types of models, such as those which assume a long--tailed distribution of trait change \citep[e.g.][]{Landis2012} or state--dependent diversification models \citep[e.g.][]{Bokma2008, Bokma2010, FitzJohn2010}. And it is not applicable to models of discrete character evolution, such as the Mk model \citep{Pagel1994, Lewis2001} and variations thereof \citep[e.g.][]{Maddison2007, WagnerMarcot2010, Mazeralli2012, Beaulieu2013} \citep[but see Felsenstein's threshold model;][]{Felsenstein2005, Felsenstein2012}. Methods to evaluate the adequacy of discrete character models are also sorely needed as the issue is just as pertinent in this case \citep{ReadNee, MaddisonFitzJohn}, though the statistics involved will necessarily be different.

%% Paragraph on measurement error
Model misspecification is not the only source of error when fitting trait models. Measurement error...

%% Paragraph on sampling bias
Another complication which has the potential to mislead inferences is based sampling of traits \citep{Freckletoninaction, longeuroname, FitzJohnwoody}. All the trait models considered in this study assume that evolution has occured independently on each branch, such that as long as the taxa included in the phylogeny are a more--or--less random sample, the parameter estimates should not be biased as a result \citep{PennellHarmon}. However, in many cases \citep[especially when using large trait databases][]{FitzJohnwoody} this will not be the case. Importantly this will not be detected by the model adequacy approach presented in this paper: the bias in the parameter estimates will be systematically the same across all the simulated datasets.

The adequacy of trait evolutionary models has largely been an ``unknown unknown'', to adopt Rumsfeld's taxonomy. That is, most research in the field has proceeded without much consideration of whether the model employed actually fits the data. We hope that our approach can help move this to the ``known known'' or at least to the ``known unknown'' category --- that is to say, we may not know why our model is not capturing the variation in the data, but at least we will know it is not.

%% Section on extensions to the method

\section{A note on implementation}

The method described here has been implemented in an R package \texttt{arbutus} It is available on CRAN and at \texttt{www.github.com/mwpennell/arbutus} and will soon be incorporated into the \texttt{Arbor} comparative methods workflow software \citep{Arbor}. For this project, we have also adopted code from the \texttt{ape} \citep{ape}, \texttt{multicore} \citep{multicore} and \texttt{ggplot2} \citep{ggplot2} repositories. We have written functions to parse the output of a number of different programs for fitting trait evolution models (see \ref{supp-table-fxns}). As this is approach was developed to be general, we have written the code in such a way that users can include their own summary statistics and trait models in the analyses; we include demonstrations of how this can be done in the supplementary material. 

\section{Concluding remarks}

While we describe this as a novel approach much of it steams directly from earlier work in the field. In the 1980s and 1990s --- the pioneering days of comparative biology --- much discussion surrounded the appropriateness of various methods and models [cite lots including some missteps, e.g. Westoby 1995]. However, as phylogenetic approaches have become more widely accepted and applied in a variety of fields, including community ecology \citep{Green1993, AckerlyDonoghue1995, Webb2002, CB2009, PennellHarmon}, behavioral ecology \citep{something}, paleobiology \citep{something else}, etc., much of this debate has seemingly been largely forgotten \citep[but see][]{Losos2010}. However, as the scale of phylogenetic comparative data has grown, these concerns regarding model adequacy are more pertinent than ever. We argue that evaluating the fit of macroevolutionary models ought to be an important component of any comparative analysis. The approach we develop in this paper can generally be applied to a variety of continuous trait models. Alternative approaches that can apply to discrete models (binary, multistate and ordinal) as well as continuous trait models which are not multivariate normal are needed. If nothing else, we hope that our article spurs renewed interest in understanding when models capture the information we are interested in. \citet[][p. 14]{Felsenstein1985} wrote that ``[p]hylogenies are fundamental to comparative biology; there is no doing it without taking them into account''. We agree, but would add that models are also fundamental to comparative biology and that there is no doing it without considering their adequacy.


\section{Acknowledgments}

We would like to thank the members of the Tempo and Mode of Trait Evolution Working Group at the National Evolutionary Synthesis Center (NESCent) as well as NESCent for funding our group. Josef Uyeda and Jeremy Beaulieu also provided insightful comments on this project. MWP was supported by a Bioinformatics and Computational Biology Graduate Fellowship from the University of Idaho and a NSERC post--graduate fellowship. LJH was supported by NSF (AVAToL #).



\newpage
\bibliographystyle{sysbio}
\bibliography{model_ad.bib}

\end{document}
